msteams-platform/tabs/how-to/using-teams-client-sdk.md

```diff
@@ -1,72 +1,469 @@
+
+The remainder of this article will walk you through the structure and latest updates to the Teams JavaScript client SDK.
+
+### Microsoft 365 support (running Teams apps in Office and Outlook)
+
+TeamsJS v.2.0 introduces the ability for certain types of Teams apps to run across the Microsoft 365 ecosystem. Currently, other Microsoft 365  application hosts (including Office and Outlook) for Teams apps support a subset of the application types and capabilities you can build for the Teams platform. This support will expand over time. For a summary of host support for Teams apps, see [Extend Teams apps across Microsoft 365](../../m365-apps/overview.md).
+
+The following table lists Teams tabs and dialogs (task modules) capabilities (public namespaces) with expanded support to run in other Microsoft 365 hosts.
+
+> [!TIP]
+> Check for host support of a given capability at runtime by calling the `isSupported()` function on that capability (namespace).
+
+|Capability | Host support | Notes |
+|-----------|--------------|-------|
+| app | Teams, Outlook, Office | Namespace representing app initialization and lifecycle. |
+| appInitialization| | Deprecated. Replaced by `app` namespace. |
+| appInstallDialog | Teams||
+| authentication | Teams, Outlook, Office | |
+| calendar | Teams, Outlook ||
+| call | Teams||
+| chat |Teams||
+| dialog | Teams, Outlook, Office | Namespace representing dialogs (formerly named *task modules*. See notes on [Dialogs](#dialogs). |
+| location |Teams| See notes on [App permissions](#app-permissions).|
+| mail | Outlook||
+| media |Teams| See notes on [App permissions](#app-permissions).|
+| pages | Teams, Outlook, Office | Namespace representing page navigation. See notes on [Deep linking](#deep-linking). |
+| people |Teams||
+| settings || Deprecated. Replaced by `pages.config`.|
+| sharing | Teams||
+| tasks | | Deprecated. Replaced by `dialog` capability. See notes on [Dialogs](#dialogs).|
+| teamsCore | Teams | Namespace containing Teams-specific functionality.|
+| video | Teams||
+
+#### App permissions
+
+App capabilities that require the user to grant [device permissions](../../concepts/device-capabilities/device-capabilities-overview.md) (such as *location* and *media*) aren't yet supported for apps running outside of Teams. There is currently no way to check app permissions in Settings or your app header when running in Outlook or Office. If a Teams app running in Office or Outlook calls a TeamsJS (or HTML5) API that triggers device permissions, that API will throw an error and fail to display a system dialog asking for user consent.
+
+Current guidance for now is to modify your code to catch the failure:
+
+* Check [isSupported()](#differentiate-your-app-experience) on a capability before using it
```

Comment:
There are a couple namespaces within the SDK that haven't been upgraded to "capabilities", which just means that there is no `isSupported` function within that namespace. In particular: `media`, `meeting` (both public) and `files` (private) do not have `isSupported implementations. So I'm wondering if this is a little misleading since `media` is an example called out above. 

Basically the user has to know that those 3 namespaces just won't work in hosts other than Teams, but they can't check at a runtime like the other capabilities.

Reply:
Added this note:

* Check [isSupported()](#differentiate-your-app-experience) on a capability before using it. `media`, `meeting`, and `files` do not yet support *isSupported* calls and do not yet work outside of Teams.